src/System.Text.Json/System/Text/Json/JsonWriter.cs

```diff
@@ -6,193 +6,592 @@
 
 namespace System.Text.Json
 {
-    public struct JsonWriter<TFormatter> where TFormatter: ITextOutput
+    public struct JsonWriter
```

Comment:
I tried to switch back to using the extension methods that exist for ITextOutput (i.e. the way it was before). This adds about a ~15% overhead.